### PR TITLE
Updates the Transform Service to accept Python extra packages through the Java API

### PR DESCRIPTION
**Pull request by chamikaramj**
Also performs following fixes.

* Transform service controller waits for expansion services to start before performing expansion or discovery operations.
*  When expansion requests fail, transform service controller sends back a detailed error with errors from all expansion services instead of just sending the last failure.

Issue is https://github.com/apache/beam/issues/26833.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by chamikaramj** [comment]
R: @Abacn

File: sdks/java/transform-service/src/main/java/org/apache/beam/sdk/transformservice/ExpansionService.java [hunk 1]
```diff
@@ -48,10 +61,61 @@ public class ExpansionService extends ExpansionServiceGrpc.ExpansionServiceImplB
...
+
+    outer:
+    for (Endpoints.ApiServiceDescriptor endpoint : endpoints) {
+      long start = System.currentTimeMillis();
+      long duration = 10;
+      while (System.currentTimeMillis() - start < SERVICE_CHECK_TIMEOUT) {
+        try {
+          String url = endpoint.getUrl();
+          int portIndex = url.lastIndexOf(":");
+          int port = portIndex > 0 ? Integer.parseInt(url.substring(portIndex + 1)) : 80;
```
Comment (Abacn) [inline sdks/java/transform-service/src/main/java/org/apache/beam/sdk/transformservice/ExpansionService.java:79]: Is default port 80 documented somewhere / rely on some other components ? If I remember correctly the port for expansion service is dynamically signed?

**Review by Abacn — COMMENTED** [review]
Thanks, the change looks good - noticed that the unit test is not get exercised

**Review by chamikaramj — COMMENTED** [review]
Thanks. PTAL.

**Comment by chamikaramj** [comment]
Run Java PreCommit

**Comment by chamikaramj** [comment]
Run Python_Transforms PreCommit

File: sdks/java/transform-service/src/main/java/org/apache/beam/sdk/transformservice/ExpansionService.java (hunk 1 above)
Comment (chamikaramj) [reply sdks/java/transform-service/src/main/java/org/apache/beam/sdk/transformservice/ExpansionService.java]: Thanks. Updated.

**Review by Abacn — APPROVED** [review]
Thanks, LGTM

before merge, check that github action jobs are also completed - #28909

**Comment by chamikaramj** [comment]
Run Java PreCommit

**Comment by chamikaramj** [comment]
Run TransformService_Direct PostCommit

**Comment by chamikaramj** [comment]
Run Portable_Python PreCommit

**Comment by chamikaramj** [comment]
Run Python_PVR_Flink PreCommit

**Comment by chamikaramj** [comment]
Thanks. All tests are passing.

**Pull request merged**